crates/bevy_pbr/src/render/pbr_functions.wgsl

```diff
@@ -201,15 +205,24 @@ fn pbr(
     let reflectance = in.material.reflectance;
     let F0 = 0.16 * reflectance * reflectance * (1.0 - metallic) + output_color.rgb * metallic;
 
-    // Diffuse strength inversely related to metallicity
-    let diffuse_color = output_color.rgb * (1.0 - metallic);
+    // Diffuse strength is inversely related to metallicity, specular and diffuse transmission
+    let diffuse_color = output_color.rgb * (1.0 - metallic) * (1.0 - transmission) * (1.0 - diffuse_transmission);
+
+    // Diffuse transmissive strength is inversely related to metallicity and specular transmission, but directly related to diffuse transmission
+    let diffuse_transmissive_color = output_color.rgb * (1.0 - metallic) * (1.0 - transmission) * diffuse_transmission;
+
+    // Calculate the world position of the second Lambertian lobe used for diffuse transmission, by subtracting material thickness
+    let diffuse_transmissive_lobe_world_position = in.world_position - vec4<f32>(in.world_normal, 0.0) * thickness;
```

Comment:
Could we run into self-intersection issues here, if the thickness is set too high? Do we care if so?

Reply:
Yeah, especially for transmitted shadows it's a _big_ issue, which is why I had to add `NotTransmittedShadowReceiver`. For the actual shading it's not a big deal, as the second lambertian lobe is just an approximation of the real deal anyway.